core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -137,6 +137,7 @@ export class PickerColumnInternal implements ComponentInterface {
        * is actively visible to the user.
        */
       this.scrollActiveItemIntoView();
+      this.focusActiveItem();
```

Comment:
Focusing the active picker column item on value change will cause the screen reader to accurately announce the value change.